Issue 24 lz4 frame module

tzulberti-jampp opened the pull request:
Solve #24

tzulberti-jampp committed:
ISSUE-24: Solve issue when usign lz4 0.18

tzulberti-jampp committed:
ISSUE-24: Revert removing trailing spaces

tzulberti-jampp committed:
ISSUE-24: Revert removing trainling space

klaussfreire reviewed (changes requested):
Nope. They're not interchangeable. A change like this needs far more care and at least a minor bump. Also a version change in the file format, and compatibility code when reading the old format.

tzulberti-jampp commented:
If I understand correctly the version bump is required because what the file created with lz4 0.8.1 is going to be imcompatible with the one created with version 0.18.1. Not because of the difference on code...

I googled who to make lz4 compatible between versions but I wasn't able to find any information about this. Do you have an idea on who to make those versions compatible?

klaussfreire commented:
Yes. Use the file format version embedded in the bundles and have both versions of the code.

klaussfreire commented:
It will probably be more efficient to talk about it in person tomorrow

tzulberti-jampp committed:
ISSUE-24: use block module instead of frame

It seems that the 0.18.1 block module is compatible with the
compress/decompress version used on previous versions (0.8, 0.9, 0.10)

klaussfreire commented on sharedbuffers/mapped_struct.py:
```diff
@@ -298,8 +297,15 @@ def unpack_from(cls, buf, offs, idmap = None, array = array.array, itemsizes = {
-lz4_compress = cython.declare(object, lz4.compress)
+try:
+    import lz4.block
+    lz4_functions_module = lz4.block
```
lz4_block?

tzulberti-jampp committed:
ISSUE-24: Do not use a variable to reference de module

tzulberti-jampp committed:
Merge branch 'master' into ISSUE-24-Lz4_frame_module

zeitos reviewed (approved):
LGTM
